standards.md

```diff
@@ -160,33 +194,15 @@ what we want.
 * If possible, in addition to code snippets, include a reference to an end to end example
 * Ensure that all links and references are valid
 
-## Functionality
```

Comment:
I would suggest keeping this section -- we do want to always be able to cut a release and should avoid merging a PR that puts us in an unreleasable state.

For incremental addition of new features, the incremental changes need to be made in a way that is non-breaking and non-blocking -- such as flag guards and the like.

Reply:
I added a note that new features and api fields should be gated behind feature flags.
I agree we always want to be in an unreleasable state, but having this guidance in the code standards has caused some confusion since it's not clear how we'd get into an "unreleasable" state or what that actually means (thanks @XinruZhang for bringing this up!). I think it may be better to have specific guidance on what types of PRs to avoid. WDYT?

Reply:
I'm not sure about removing this. 

I think PR authors should ask themselves the question "if this was the last PR before the release, would we produce a usable release?". This is important to make sure that our nightly builds are usable as well as it helps very much in reducing the complexity the release manager may have to deal with.

Of course, CI takes a big role in ensuring that the code can be compiled and is generally safe and functionally sound. Feature flags help with incomplete features, but also documentation should be written so that it does not mislead users. We can try to make a complete list of all the things to consider, but I would prefer we still ask authors to ask themselves the question about the "releasable" state.

Reply:
A few affirmative guidelines and / or antipatterns that I can think of:

- No PR should require concurrent release of code in another repository
- No PR should require that a release be held pending some later PR
- No PR should include a breaking change without either (1) a controlling feature flag or (2) a release note that explains the intentionality of the breaking change
- A PR that intentionally changes functionality should only be merged along with compatible, passing tests in the same PR

Reply:
Thanks for the discussion here! I brought back the note about "unreleasable state" with a bit more explanation. @bendory I included all your examples but changed the third to "PRs must adhere to the project's API stability policy." I think it would be good as a follow up to move most of the contents of https://github.com/tektoncd/pipeline/blob/main/api_compatibility_policy.md that aren't specific to pipelines into the community repo so that all projects can link them.

Reply:
/LGTM

modulo one minor nit in a separate comment